src/targets/awsLambdaLayer.ts

```diff
@@ -0,0 +1,202 @@
+import * as fs from 'fs';
+import { logger as loggerRaw } from '../logger';
+import { TargetConfig } from '../schemas/project_config';
+import { BaseTarget } from './base';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { ConfigurationError, reportError } from '../utils/errors';
+import {
+  AddLayerVersionPermissionRequest,
+  Lambda,
+  PublishLayerVersionCommandOutput,
+  PublishLayerVersionRequest,
+} from '@aws-sdk/client-lambda';
+import { DescribeRegionsCommandOutput, EC2 } from '@aws-sdk/client-ec2';
+
+const logger = loggerRaw.withScope(`[aws-lambda-layer]`);
+
+/** Config options for the "aws-lambda-layer" target. */
+interface AwsLambdaTargetOptions extends TargetConfig {
+  /** AWS access key ID, set as AWS_ACCESS_KEY_ID. */
+  awsAccessKeyId: string;
+  /** AWS secret access key, set as `AWS_SECRET_ACCESS_KEY`. */
+  awsSecretAccessKey: string;
+}
+
+/**
+ * Target responsible for uploading files to AWS Lambda.
+ */
+export class AwsLambdaLayerTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'aws-lambda-layer';
+  /** Target options */
+  public readonly awsLambdaConfig: AwsLambdaTargetOptions;
+  /** Name of the layer to be published */
+  static layerName: string;
+  /** All AWS regions available for the current user */
+  public awsRegions: any;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.awsLambdaConfig = this.getAwsLambdaConfig();
+  }
+
+  /**
+   * Extracts AWS Lambda target options from the environment.
+   */
+  protected getAwsLambdaConfig(): AwsLambdaTargetOptions {
+    if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+      throw new ConfigurationError(
+        `Cannot publish AWS Lambda Layer: missing credentials.
+        Please use AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables.`
+      );
+    }
+    return {
+      awsAccessKeyId: process.env.AWS_ACCESS_KEY_ID,
+      awsSecretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    };
+  }
+
+  /**
+   * Requests all regions that are enabled for the current account (or all
+   * regions) to AWS. For more information, see
+   * https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EC2.html#describeRegions-property
+   */
+  public async getAwsRegions(): Promise<DescribeRegionsCommandOutput> {
+    const ec2 = new EC2({ region: 'us-east-2' });
```

Comment:
Could use `AWS_DEFAULT_REGION` like aws-cli; can be left for a future improvement.